matrix/SquareMatrix.hpp

```diff
@@ -118,6 +130,119 @@ class SquareMatrix : public Matrix<Type, M, M>
         }
         return res;
     }
+
+    // zero all offdiagonal elements and keep corresponding diagonal elements
+    template <size_t width>
```

Comment:
Generally template arguments are capitalized, at least for the first letter, to make it easier to differentiate from other variables